### ISSUE TITLE
Support For php56u-pecl-krb5

### ISSUE THREAD

**Issue by jmkeyes** [iuscommunity/wishlist#18]
I've created a package for the [PHP 5.6 Kerberos5 extension](https://github.com/jmkeyes/php56u-pecl-krb5) and I've been using it in a development sandbox privately, in conjunction with the other IUS-provided PHP packages. Please let me know if it requires any changes; I'd be happy to test them.

**Comment by carlgeorge** [comment]
Howdy @jmkeyes,

I have a few questions/suggestions.

1. Did you write this spec file from scratch or did you base it on another?  The changelog indicates that it is a brand new spec, but if it was based on another then the previous changelog entries should be included.  Typically we fork from Fedora spec files and leave the prior changelog entries intact.

2. Fedora includes a [patch](http://pkgs.fedoraproject.org/cgit/php-pecl-krb5.git/tree/krb5-build.patch) related to [this](http://svn.php.net/viewvc?view=revision&revision=333127).  Do you think it is relevant/necessary for an IUS package?  From a quick glance I'm not sure whether we need it or not, hopefully you have more insight into the purpose than I can grok from the minimal context given there.

3. Best practice is to have virtual provides have a version and release, i.e. `Provides: php-%{pecl_name} = %{version}-%{release}` instead of just `Provides: php-%{pecl_name} = %{version}`.

**Comment by jmkeyes** [comment]
1. If I remember correctly, I based it on both the `php56u-pecl-mongo` and `php56u-pecl-jsonc` spec files from IUS and didn't reference the upstream spec files. The upstream probably has a better spec file for this purpose.
2. I didn't look into any patches when I built the RPM originally as the stock version compiled and worked for the version of PHP that I was testing with. That patch appears to correct for a compilation failure on certain versions of PHP where a `#define` changed to an `enum` as of PHP 5.5. It'd probably be a good idea to include it.
3. I've appended the release version as requested. (See commit 8064c72.)

**Comment by carlgeorge** [comment]
I definitely want to accept your contributions, but I would like to start the package as a fork of the Fedora package, including the Fedora changelog and that patch.  How about I start the repo, import the Fedora SRPM, and then you send us a pull request that brings in the IUS-isms from your spec file?  You can break it down into as many commits as you see appropriate.  You can observe this workflow in some of our other packages.

* https://github.com/iuscommunity-pkg/php56u-pecl-gearman/commits/master
* https://github.com/iuscommunity-pkg/php56u-xcache/commits/master

**Comment by jmkeyes** [comment]
I understand. I've renamed the `master` branch of my repository to `deprecated` and started an orphan `master` branch in it's place. I'll look into rebuilding the spec file from Fedora rawhide and publish my modifications after testing them. If all goes well, I will have a candidate package for review by the end of this weekend.

**Comment by carlgeorge** [comment]
Looks good so far.  One quick thing I noticed in the new master branch is that the macro `ini_name` is not defined.  While porting you removed this Fedora conditional statement:

```
%if "%{php_version}" < "5.6"
%global ini_name    %{pecl_name}.ini
%else
%global ini_name    40-%{pecl_name}.ini
%endif
```

The conditional is unnecessary, but we still need the contents of the else condition so later references to `ini_name` work.  Please add this one line back to the spec file.

```
%global ini_name    40-%{pecl_name}.ini
```

**Comment by carlgeorge** [comment]
Two more small things.  We don't need to carry forward the Fedora release number, since this is our first version of the package.  Change the release from `7.ius%{?dist}` to `1.ius%{?dist}`.  Also, in the changelog entry you indicated an epoch of 1, but there isn't one.  Probably just a copy/paste error from those examples I gave you.  Change the changelog version number from `1:1.0.0-7.ius` to `1.0.0-1.ius`.

**Comment by jmkeyes** [comment]
The requested changes have been made.

**Comment by carlgeorge** [comment]
In the past we have forked contributors repos, but that has side effects within GitHub's search.  I have "transferred" your repo to our organization, using `git clone` and then `git remote` in order to preserve your commits.

https://github.com/iuscommunity-pkg/php56u-pecl-krb5

Going forward we will maintain it from this repo.  I have completed some initial builds, and they will be available in the testing repos tonight after our automation runs.  Give them a shot and let me know how they work for you.
